.upstream-tests/test/cuda/proclaim_return_type.fail.cpp

```diff
@@ -0,0 +1,29 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++98, c++03
+// UNSUPPORTED: !nvcc
+
+#include <cuda/functional>
+
+#include <assert.h>
+
+int main(int argc, char ** argv)
+{
+#ifdef __CUDA_ARCH__
+  auto f = cuda::proclaim_return_type<double>(
```

Comment:
I had to do some digging, but I found it.

`// expected-error{{no type named 'x' in 'Foo'}}`

You can use this to make .fail tests check for a *specific* diagnostic.

Reply:
Awesome, thanks!

Reply:
@wmaxey I don't think that `expected-error` affects anything in libcu++ tests. I tried to find how lit handles this annotation, but it doesn't. It's a clang's verifier [feature](https://clang.llvm.org/doxygen/classclang_1_1VerifyDiagnosticConsumer.html). The `expected-*` annotation is only handled by clang in a special mode `clang -cc1 -verify source.cpp`. This options can't be passed as `nvcc -ccbin clang -Xcompiler=...` because they conflict with different command line arguments that nvcc uses internally. Different concern is related to the fact that expected errors are not errors. So even if we manage to use `-cc1 -verify` somehow, the fail test will pass, which is not expected. 

Am I missing something?

Reply:
I don't quite recall. I vaguely remember editing a few lines that were checking for expected errors where a diagnostic was different on NVCC. I don't think it's necessary, but if I find it I'll try and document it for next time.